slatedb/src/wal_id.rs

```diff
@@ -0,0 +1,3 @@
+pub(crate) trait WalIdStore: Send + Sync + 'static {
```

Comment:
i was considering use a `MononicSeq` to manage the last wal id, but it turns out has a lot of footprints in the code base.

i think this trait is a temporary work around to avoid wal_buffer to drectly depends on `CoreDbState`. i hope i could remove it in another pr.